tests/apps/test_graph_max_clique.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2019 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Unit tests for strawberryfields.apps.graph.max_clique
+"""
+# pylint: disable=no-self-use,unused-argument
+import pytest
+import networkx as nx
+import numpy as np
+
+from strawberryfields.apps.graph import max_clique
+from strawberryfields.apps.graph import resize
+
+
+def patch_random_choice(x):
+    """Dummy function for ``np.random.choice`` to make output deterministic. This dummy function
+    just returns the element of ``x`` specified by ``element``."""
+    if isinstance(x, int):  # np.random.choice can accept an int or 1D array-like input
+        return 0
+
+    return x[0]
+
+
+def patch_clique_resize(clique, graph, node_select):
+    """Dummy function for ``clique_grow`` and ``clique_swap`` to make output unchanged."""
+    return clique
+
+
+class TestLocalSearch:
+    """Tests for the function ``max_clique.local_search``"""
+
+    def test_bad_iterations(self):
+        """Test if function raises a ``ValueError`` when a non-positive number of iterations is
+        specified"""
+        with pytest.raises(ValueError, match="Number of iterations must be a positive int"):
+            max_clique.local_search(clique=[0, 1, 2, 3], graph=nx.empty_graph(5), iterations=-1)

```

Comment:
Is it not a problem that `clique=[0, 1, 2, 3]` is not actually a clique? Doesn't it make more sense to use `graph=nx.complete_graph(5)?`